### PR TITLE
feat: shed: blockstore/vlog to car export cmds

### PR DESCRIPTION
**Pull request by magik6k**
Just some tools to get data out of big badger datastores, and into car files.

### CODE REVIEW

File: cmd/lotus-shed/datastore-vlog.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+)
+
+var datastoreVlog2CarCmd = &cli.Command{
+	Name:  "vlog2car",
+	Usage: "convert badger blockstore .vlog to .car",
+	Flags: []cli.Flag{
+		&cli.PathFlag{
+			Name:     "vlog",
```
Comment (ZenGround0) [inline cmd/lotus-shed/datastore-vlog.go:27]: For my own curiosity: what is a badger vlog?  I see a bunch of `.vlog` files under datastore repos.  What is the meaning of each vlog subunit?  Is it a partition of data based on the key for example?

File: cmd/lotus-shed/datastore-vlog.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+	block "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-base32"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+)
+
+var datastoreVlog2CarCmd = &cli.Command{
+	Name:  "vlog2car",
+	Usage: "convert badger blockstore .vlog to .car",
```
Comment (ZenGround0) [inline cmd/lotus-shed/datastore-vlog.go:24]: Any warnings needed about the daemon needing to be turned off as in the following utility?

Comment (ZenGround0) [reply cmd/lotus-shed/datastore-vlog.go:24]: I'm not seeing any interaction with the repo lock file, any chance for problems if the node is online?

**Review by ZenGround0 — APPROVED** [review]
I don't know badger internals so this review is fairly shallow with only some high level fearmongering about repo lock safety.  Even without knowing badger the general idea of car to badger seems *really* nice.

File: cmd/lotus-shed/datastore-vlog.go (hunk 2 above)
Comment (magik6k) [reply cmd/lotus-shed/datastore-vlog.go:24]: vlog files are append-only, so that's (probably) fine

File: cmd/lotus-shed/datastore-vlog.go (hunk 1 above)
Comment (magik6k) [reply cmd/lotus-shed/datastore-vlog.go:27]: It's basically an append-only (afaiu) k-v write log. The `Entry`/`header` structs below tell you more exactly what is in those files

**Pull request merged**